src/compiler/checker.ts

```diff
@@ -3993,6 +4027,37 @@ namespace ts {
             return links.resolvedType;
         }
 
+        // We do not perform supertype reduction on intersection types. Intersection types are created only by the &
+        // type operator and we can't reduce those because we want to support recursive intersection types. For example,
+        // a type alias of the form "type List<T> = T & { next: List<T> }" cannot be reduced during its declaration.

```

Comment:
We could consider that, but given a type A and a subtype B, the only observable difference between A & B and B would be when B is a primitive type (because expression operators would then come into effect), and use of intersection types with primtives isn't really a scenario. So, it would rarely if ever make a meaningful difference.